nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -202,11 +213,11 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final boolean addFragmentAttrs;
         switch (packagingFormat) {
             case TAR_FORMAT:
```

Comment:
yeah I noticed the same thing when I added the filter ability. I didn't want to make any major refactoring to it in fear it would break. I could give it a try though, because I agree, it doesn't make sense to have this logic for each incoming flowfile.